networkx/algorithms/components/weakly_connected.py

```diff
@@ -158,7 +160,7 @@ def is_weakly_connected(G):
             """Connectivity is undefined for the null graph."""
         )
 
-    return len(next(weakly_connected_components(G))) == len(G)
+    return sum(1 for node in _plain_bfs(G, arbitrary_element(G))) == len(G)
```

Comment:
What's the motivation here? I prefer the original one for readability and it's not obvious to me that the latter would generally improve performance.

Reply:
before it executed 

```
    seen = set()
    for v in G:
        if v not in seen:
            c = set(_plain_bfs(G, v))
            seen.update(c)
            yield c
```

unnecessarily, but maybe something like `len(set(_plain_bfs(G, arbitrary_element(G))))` could be better anyway?

On the other hand, as you pointed for consistency would be better to return sets on both, so if we do this way we can use the way in the other file

Reply:
updated so that both returns a set, this way we can also use the same code in both methods, there is a null graph test which is failing though

Reply:
solved the problem with the test failing, what do you think about doing it this way?